wsd/ClientSession.cpp

```diff
@@ -2693,347 +2693,367 @@ bool ClientSession::handleKitToClientMessage(const std::shared_ptr<Message>& pay
             }
 
-            // Now terminate.
-            docBroker->closeDocument("extracteddocumentstructure");
-            return true;
-        }
-        else if (tokens.equals(0, "transformeddocumentstructure:"))
-        {
-            LOG_TRC("Sending transformed document structure response.");
-            if (!saveAsSocket)
-                LOG_ERR("Error in transformeddocumentstructure: not in isConvertTo mode");
-            else
+            if (error)
             {
-                http::Response httpResponse(http::StatusCode::OK);
-                FileServerRequestHandler::hstsHeaders(httpResponse);
-                httpResponse.set("Last-Modified", Util::getHttpTimeNow());
-                httpResponse.set("X-Content-Type-Options", "nosniff");
-                httpResponse.setBody(payload->jsonString(), "application/json");
+                http::Response httpResponse(http::StatusCode::InternalServerError);
+                httpResponse.setContentLength(0);
                 saveAsSocket->sendAndShutdown(httpResponse);
             }
-
-            // Now terminate.
-            docBroker->closeDocument("transformeddocumentstructure");
-            return true;
         }
-        else if (tokens.equals(0, "sendthumbnail:"))
-        {
-            LOG_TRC("Sending get-thumbnail response.");
-            if (!saveAsSocket)
-                LOG_ERR("Error in sendthumbnail: not in isConvertTo mode");
-            else
-            {
-                bool error = false;
-
-                if (firstLine.find("error") != std::string::npos)
-                    error = true;
-
-                if (!error)
-                {
-                    int firstLineSize = firstLine.size() + 1;
-                    std::string thumbnail(payload->data().data() + firstLineSize, payload->data().size() - firstLineSize);
-
-                    http::Response httpResponse(http::StatusCode::OK);
-                    FileServerRequestHandler::hstsHeaders(httpResponse);
-                    httpResponse.set("Last-Modified", Util::getHttpTimeNow());
-                    httpResponse.set("X-Content-Type-Options", "nosniff");
-                    httpResponse.setBody(std::move(thumbnail), "image/png");
-                    saveAsSocket->sendAndShutdown(httpResponse);
-                }
 
-                if (error)
-                {
-                    http::Response httpResponse(http::StatusCode::InternalServerError);
-                    httpResponse.setContentLength(0);
-                    saveAsSocket->sendAndShutdown(httpResponse);
-                }
-            }
-
-            docBroker->closeDocument("thumbnailgenerated");
-        }
-    }
-    else
-    {
-        LOG_INF("Ignoring notification on password protected document: " << firstLine);
+        docBroker->closeDocument("thumbnailgenerated");
     }
 
-    // Forward everything else.
-    return forwardToClient(payload);
+    return std::nullopt;
```

Comment:
A comment that this is "fall-through" case that isn't handled, and the message needs to be forwarded, is a good idea. Something like: `// Fall-through, as we couldn't handle the message.`